### PR TITLE
Update Ruby Dependencies

### PR DESCRIPTION
**Pull request by ahangarha**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1595)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 2
 
 aliases:
   - &docker-image
-    - image: circleci/ruby:2.7-node-browsers
+    - image: circleci/ruby:3.0-node-browsers
```
Comment (justin808) [inline .circleci/config.yml:6]: is 3.0 the newest

File: Gemfile.development_dependencies [hunk 2]
```diff
@@ -1,16 +1,15 @@
 # frozen_string_literal: true
 
-gem "shakapacker", "7.0.1"
+gem "shakapacker", "7.2.1"
 gem "bootsnap", require: false
-gem "rails", "~> 7.0", ">= 7.0.1"
+gem "rails", "~> 7.1"
```
Comment (justin808) [inline Gemfile.development_dependencies:5]: any updates needed for the Rails upgrade?

File: .circleci/config.yml (hunk 1 above)
Comment (ahangarha) [reply .circleci/config.yml:6]: Do we use circleci for ROR at all?

File: Gemfile.development_dependencies (hunk 2 above)
Comment (ahangarha) [reply Gemfile.development_dependencies:5]: I didn't notice anything to apply to this project.

**Review by justin808 — APPROVED** [review]
LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (justin808) [reply .circleci/config.yml:6]: No, but good to have the example file.

**Pull request merged**